native/common/include/jpype.h

```diff
@@ -101,11 +88,16 @@ using std::list;
 /** Definition of commonly used template types */
 typedef vector<string> StringVector;
 
+typedef jvalue (*jconverter)(char*) ;
+
+extern jconverter getConverter(const char* from, int itemsize, const char* to);
```

Comment:
So the type conversion for an array happens within a Java class? I think this deserves some comment to be easily understandable.

Reply:
When making a buffer transfer from one buffered type to another (numpy to Java array) where the format code does not match I have to get a function which does the conversion.  I originally had this split out by the type requested and such but that was really awkward.  Thus I just created one giant autogenerated file that returns every combination that we could need.  Then the getConverter looks up the function and passes it back so we can use it over and over for each item in the list.

This is a big enhancement over the reference code in CPython that I was looking at.  For CPython implementation they do a giant switch statement with a comment (paraphrasing) "Don't touch this or it will have huge speed changes as gcc does a lookup table".  As they go through the table for each item to convert I am not sure why they don't just allocate a function for each conversion and then go through the switch once like I did.

Reply:
addressed.